src/main/java/mate/controller/CreateCarController.java

```diff
@@ -0,0 +1,41 @@
+package mate.controller;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Manufacturer;
+import mate.service.CarService;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/cars/add")
+public class CreateCarController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate");
+    private final CarService carService = (CarService) injector.getInstance(CarService.class);
+    private final ManufacturerService manufacturerService =
+            (ManufacturerService) injector.getInstance(ManufacturerService.class);
+
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/create_car.jsp").forward(req, resp);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String model = req.getParameter("model");
+        Manufacturer manufacturer = manufacturerService.get(Long.valueOf(req.getParameter("id")));
+        Car car = new Car();
+        car.setManufacturer(manufacturer);
+        car.setModel(model);
+        car.setDrivers(new ArrayList<>());
+        carService.create(car);
```

Comment:
As I understand, you are going just to save this car in DB, and service or dao will not add any drivers before saving. So, why do we need this empty list here? Can it be somehow stored in the DB?

Reply:
Якщо я прибираю 37 строку - car.setDrivers(new ArrayList<>()) , виникає 500 помилка Internal Server Error
Message Cannot invoke "java.util.List.size()" because "drivers" is null
Description The server encountered an unexpected condition that prevented it from fulfilling the request.
Exception
java.lang.NullPointerException: Cannot invoke "java.util.List.size()" because "drivers" is null
	mate.dao.CarDaoImpl.insertAllDrivers(CarDaoImpl.java:164)
	mate.dao.CarDaoImpl.create(CarDaoImpl.java:42)
	mate.dao.CarDaoImpl.create(CarDaoImpl.java:19)
	mate.service.CarServiceImpl.create(CarServiceImpl.java:34)
	mate.service.CarServiceImpl.create(CarServiceImpl.java:10)
	mate.controller.CreateCarController.doPost(CreateCarController.java:38)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:681)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)